test/integ/test_cursor.py

```diff
@@ -1690,3 +1690,138 @@ def test_decoding_utf8_for_json_result(conn_cnx):
     mock_resp.content = "À".encode("latin1")
     with pytest.raises(Error):
         result_batch._load(mock_resp)
+
+
+@pytest.mark.skipolddriver
+def test_switch_nanoarrow_and_vendored_arrow(conn_cnx, caplog):
+    from snowflake.connector.cursor import _get_client_nanoarrow_setting
+
+    origin_value = snowflake.connector.cursor.USE_NANOARROW_CONVERTER
+
+    # test setting directly the parameter
+
+    snowflake.connector.cursor.USE_NANOARROW_CONVERTER = None
+    with conn_cnx() as con:
+        with con.cursor() as cur, caplog.at_level(logging.DEBUG):
+            cur.execute("select 1").fetchall()
+            assert "Using nanoarrow as the arrow data converter" in caplog.text
+
+    snowflake.connector.cursor.USE_NANOARROW_CONVERTER = True
+    with conn_cnx() as con:
+        with con.cursor() as cur, caplog.at_level(logging.DEBUG):
+            cur.execute("select 1").fetchall()
+            assert "Using nanoarrow as the arrow data converter" in caplog.text
+
+    snowflake.connector.cursor.USE_NANOARROW_CONVERTER = False
+    with conn_cnx() as con:
+        with con.cursor() as cur, caplog.at_level(logging.DEBUG):
+            cur.execute("select 1").fetchall()
+            assert "Using vendored arrow as the arrow data converter" in caplog.text
+
+    snowflake.connector.cursor.USE_NANOARROW_CONVERTER = origin_value
+
+    # test setting by env var
+
+    os.environ["USE_NANOARROW_CONVERTER"] = "1"
+    assert _get_client_nanoarrow_setting() is True
+
+    os.environ["USE_NANOARROW_CONVERTER"] = "True"
+    assert _get_client_nanoarrow_setting() is True
+
+    os.environ["USE_NANOARROW_CONVERTER"] = "true"
+    assert _get_client_nanoarrow_setting() is True
+
+    snowflake.connector.cursor.USE_NANOARROW_CONVERTER = _get_client_nanoarrow_setting()
+
+    with conn_cnx() as con:
+        with con.cursor() as cur, caplog.at_level(logging.DEBUG):
+            cur.execute("select 1").fetchall()
+            assert "Using nanoarrow as the arrow data converter" in caplog.text
+
+    os.environ["USE_NANOARROW_CONVERTER"] = "False"
```

Comment:
For setting env variables please use pytest's monkeypatch's context context manager. It'll make sure to reset values back to what they were before, errors here could lead to test pollutions

Reply:
resolved in #1758